src/market/service.ts

```diff
@@ -76,7 +76,7 @@ export class MarketService {
   private demandEventListener(event: Event) {
     const proposal = (event as DemandEvent).proposal;
     const error = (event as DemandEvent).error;
-    if (error) {
+    if (error || !proposal) {
```

Comment:
Looks a bit too aggressive to me. @mgordel , what do you think? 

I would say that if the issues is that later starting from lines 84 etc `proposal is possibly undefined`, then I would wrap that section so that instead of this, we have:

```ts
 private demandEventListener(event: Event) {
    const proposal = (event as DemandEvent).proposal;
    const error = (event as DemandEvent).error;
    
    if (error) {
      this.logger?.error("Subscription failed. Trying to subscribe a new one...");
      this.resubscribeDemand().catch((e) => this.logger?.warn(e));
      return;
    }
    
    if (proposal) {
      if (proposal.isInitial()) this.processInitialProposal(proposal);
      else if (proposal.isDraft()) this.processDraftProposal(proposal);
      else if (proposal.isExpired()) this.logger?.debug(`Proposal hes expired ${proposal.id}`);
      else if (proposal.isRejected()) {
        this.proposalsCount.rejected++;
        this.logger?.debug(`Proposal hes rejected ${proposal.id}`);
      }
    }
  }
```

Reply:
Oh, and fix the typo `Proposal hes rejected ` into `has rejected` :D